akka-typed/src/main/scala/akka/typed/ActorRef.scala

```diff
@@ -44,26 +36,45 @@ abstract class ActorRef[-T] extends java.lang.Comparable[ActorRef[Any]] { this:
    * and more than one Actor instance can exist with the same path at different
    * points in time, but not concurrently.
    */
-  def path: ActorPath = untypedRef.path
+  final val path: ActorPath = _path
 
-  override def toString = untypedRef.toString
-  override def equals(other: Any) = other match {
-    case a: ActorRef[_] ⇒ a.untypedRef == untypedRef
-    case _              ⇒ false
+  /**
+   * Comparison takes path and the unique id of the actor cell into account.
+   */
+  final def compareTo(other: ActorRef[Nothing]) = {
+    val x = this.path compareTo other.path
+    if (x == 0) if (this.path.uid < other.path.uid) -1 else if (this.path.uid == other.path.uid) 0 else 1
+    else x
   }
-  override def hashCode = untypedRef.hashCode
-  override def compareTo(other: ActorRef[Any]) = untypedRef.compareTo(other.untypedRef)
+
+  final override def hashCode: Int = path.uid
+
+  /**
+   * Equals takes path and the unique id of the actor cell into account.
+   */
+  final override def equals(that: Any): Boolean = that match {
+    case other: ActorRef[_] ⇒ path.uid == other.path.uid && path == other.path
+    case _                  ⇒ false
+  }
+
+  final override def toString: String = s"Actor[${path}#${path.uid}]"
 }
 
 /**
  * This trait is used to hide the `!` method from Java code.
  */
 trait ScalaActorRef[-T] { this: ActorRef[T] ⇒
```

Comment:
Yes, good question: I just kept it from the untyped version, but since this ActorRef has even less methods than the old one, it might actually be fine. Trying it out.